polyphemus/lib/etls/ipi/ipi_etls.rb

```diff
@@ -7,13 +7,46 @@
 require_relative './linkers/ipi_rna_seq_raw_fastq_files_linker'
 require_relative './record_creators/ipi_rna_seq_and_plate_record_creator'
 require_relative '../linker_processor'
+require_relative '../shared/single_cell/single_cell_linkers'
 
 class Polyphemus
   module Ipi
+    # Broken out to simplify tests
+    module SingleCellLinkers
+      def add_single_cell_linkers!
+        process_watch_type_with(
+          bucket('data')
+            .watcher('single_cell_pool_processed')
+            .watch(/^single_cell_GEX\/.*\/[^\/]*POOL[^\/]*\/.*$/),
```

Comment:
For other projects, there wind up being like four or more different kinds of single_cell analysis. (`single_cell_GEX`, `single_cell_BCR`, `single_cell_TCR`, and `single_cell_CITE` .. perhaps with ATAC later?). Would you make different watchers in that case (line 19) to go along with the different regexes?

Reply:
It might be best to make the identifier validation for the single-cell model in magma/ipi the canonical place to retrieve your regexp (and set that regexp correctly in Magma if it is not), so the two do not go out of sync.